### ISSUE TITLE
Domogik develop don't work correctly for me

### ISSUE THREAD

**Issue by vdomos** [domogik/domogik#485]
i can start plugins in admin interface, plugins started are  "Alive"

In plugins logs, device'update is ok:

    2017-05-08 13:28:26,284 domogik-weather INFO Preparing to send value for sensor id:63, name: forecast_9_condition_code, value: 4
    2017-05-08 13:28:26,285 domogik-weather INFO Preparing to send value for sensor id:61, name: forecast_9_temperature_low, value: 13
    2017-05-08 13:28:26,285 domogik-weather INFO Data successfully sent for 599573

but devices are not updated some time in admin interface.
Domoweb don't discover new devices too.

I have errors like that in logs:

    core_admin.log:2017-05-08 13:43:51,677 domogik-admin ERROR Traceback (most recent call last):
    core_butler.log:2017-05-08 13:12:25,698 domogik-butler INFO Learn file NOT found : /var/lib/domogik/resources/butler/butler_learn.rive. This is not an error. You just have learn nothing to your butler ;)
    core_butler.log:2017-05-08 13:12:35,716 domogik-butler WARNING Error while getting the devices list over MQ. Error is : Traceback (most recent call last):
    core_butler.log:AttributeError: 'NoneType' object has no attribute 'get'
    core_butler.log:2017-05-08 13:44:32,014 domogik-butler INFO Learn file NOT found : /var/lib/domogik/resources/butler/butler_learn.rive. This is not an error. You just have learn nothing to your butler ;)
    core_butler.log:2017-05-08 13:44:42,025 domogik-butler WARNING Error while getting the devices list over MQ. Error is : Traceback (most recent call last):
    core_butler.log:AttributeError: 'NoneType' object has no attribute 'get'
    core_manager.log:2017-05-08 13:12:30,571 domogik-manager ERROR Query config for client vigipollens on host ares, key * : no response from admin
    core_manager.log:2017-05-08 13:12:40,577 domogik-manager ERROR Query config for client vigipollens on host ares, key configured : no response from admin
    core_manager.log:2017-05-08 13:12:50,589 domogik-manager ERROR Query config for client vigipollens on host ares, key auto_startup : no response from admin
    core_manager.log:2017-05-08 13:28:03,612 domogik-manager ERROR Query config for client weather on host ares, key test_mode : no response from admin
    core_manager.log:2017-05-08 13:28:13,621 domogik-manager ERROR Query config for client weather on host ares, key test_option : no response from admin
    core_manager.log:2017-05-08 13:44:36,891 domogik-manager ERROR Query config for client vigipollens on host ares, key * : no response from admin
    core_manager.log:2017-05-08 13:44:46,901 domogik-manager ERROR Query config for client vigipollens on host ares, key configured : no response from admin
    core_manager.log:2017-05-08 13:44:56,911 domogik-manager ERROR Query config for client vigipollens on host ares, key auto_startup : no response from admin
    core_xplgw.log:2017-05-08 13:44:40,291 domogik-xplgw ERROR Updating client list failed, no response from manager (try number 1/5)
    core_xplgw.log:2017-05-08 13:45:52,258 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:KeyError: '10'

domogik develop is up to date.

Same problem on 2 installations.

**Comment by fritz-smh** [comment]
Can you give us the admin full logs ?

**Comment by vdomos** [comment]
Stop Domogik
Delete all log
Restart with only 3 plugins
Only vdevice is started at startup

Here are the logs:


[domogik_logs.zip](https://github.com/domogik/domogik/files/986441/domogik_logs.zip)

**Comment by fritz-smh** [comment]
Can you start the admin in command line with the -f option and give the full stdout ?

Thanks.

**Comment by vdomos** [comment]
How I proceed ?
if I start Domogik, kill admin et start it with -f, there is issue with gunicorn:


    $ /home/dan/dmg/domogik/src/domogik/bin/admin.py -f
    2017-05-09 15:34:36,820 domogik-admin INFO Looking for launched instances of 'admin'
    2017-05-09 15:34:36,833 domogik-admin INFO Process found : dan      10143  1.3  3.4 278560 70168 ?        Sl   15:33   0:01 /usr/bin/python /usr/local/bin/gunicorn --preload --log-level debug -b 127.0.0.1:40406 -b 192.168.0.31:40406 -b 10.0.2.15:40406 -w 2 domogik.admin.application:app
    2017-05-09 15:34:36,833 domogik-admin INFO Process found : dan      10152  0.0  3.0 278560 63288 ?        S    15:33   0:00 /usr/bin/python /usr/local/bin/gunicorn --preload --log-level debug -b 127.0.0.1:40406 -b 192.168.0.31:40406 -b 10.0.2.15:40406 -w 2 domogik.admin.application:app
    2017-05-09 15:34:36,833 domogik-admin INFO Process found : dan      10153  0.0  3.0 278560 63300 ?        S    15:33   0:00 /usr/bin/python /usr/local/bin/gunicorn --preload --log-level debug -b 127.0.0.1:40406 -b 192.168.0.31:40406 -b 10.0.2.15:40406 -w 2 domogik.admin.application:app
    2017-05-09 15:34:36,833 domogik-admin INFO There are already existing processes.
    2017-05-09 15:34:36,833 domogik-admin DEBUG __del__ Single client
    2017-05-09 15:34:36,834 domogik-admin DEBUG force_leave called
    Exception SystemExit in <bound method Admin.__del__ of <__main__.Admin object at 0x7ff50520e4d0>> ignored

**Comment by fritz-smh** [comment]
this is not an issue, just remaining threads. Kill them before being able
to launch admin.

**Comment by vdomos** [comment]
Yes, not a issue, not sure kill it was the good way.

[admin_f.txt](https://github.com/domogik/domogik/files/986956/admin_f.txt)

I only found this message: "CRITICAL] WORKER TIMEOUT" in admin STDOUT

Other ERROR in log files:

    $ grep ERROR * |grep -v "BAD direction message"|grep "2017-05-09 16:2"
    core_manager.log:2017-05-09 16:21:07,981 domogik-manager ERROR Query config for client weather on host ares, key test_mode : no response from admin
    core_manager.log:2017-05-09 16:21:17,991 domogik-manager ERROR Query config for client weather on host ares, key test_option : no response from admin
    core_manager.log:2017-05-09 16:25:23,850 domogik-manager ERROR Query config for client vigipollens on host ares, key test_mode : no response from admin
    core_manager.log:2017-05-09 16:25:33,857 domogik-manager ERROR Query config for client vigipollens on host ares, key test_option : no response from admin
    core_xplgw.log:2017-05-09 16:20:57,430 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:23:29,841 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,947 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,948 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,950 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,951 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,952 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,955 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,957 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,959 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,962 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,963 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,964 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,967 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,968 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,969 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,969 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,970 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,971 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,972 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,972 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:25:34,973 domogik-xplgw ERROR Traceback (most recent call last):
    core_xplgw.log:2017-05-09 16:26:01,833 domogik-xplgw ERROR Traceback (most recent call last):


    2017-05-09 16:20:57,430 domogik-xplgw ERROR Traceback (most recent call last):
      File "/home/dan/dmg/domogik/src/domogik/bin/xplgw.py", line 851, in run
        dev = self.all_devices[str(sen['device_id'])]
    KeyError: '10'
    --
    2017-05-09 16:23:29,841 domogik-xplgw ERROR Traceback (most recent call last):
      File "/home/dan/dmg/domogik/src/domogik/bin/xplgw.py", line 851, in run
        dev = self.all_devices[str(sen['device_id'])]
    KeyError: '10'
    --
    2017-05-09 16:25:34,947 domogik-xplgw ERROR Traceback (most recent call last):
      File "/home/dan/dmg/domogik/src/domogik/bin/xplgw.py", line 850, in run
        sen = self.all_sensors[str(senid)]
    KeyError: '91'
    --
    2017-05-09 16:25:34,948 domogik-xplgw ERROR Traceback (most recent call last):
      File "/home/dan/dmg/domogik/src/domogik/bin/xplgw.py", line 850, in run
        sen = self.all_sensors[str(senid)]
    KeyError: '92'
    --
    2017-05-09 16:25:34,950 domogik-xplgw ERROR Traceback (most recent call last):
      File "/home/dan/dmg/domogik/src/domogik/bin/xplgw.py", line 850, in run
        sen = self.all_sensors[str(senid)]
    KeyError: '86'
    --

**Comment by fritz-smh** [comment]
I don't see...
Any way to get access to the server over ssh ? (we can see this over irc)

**Comment by fritz-smh** [comment]
As said in an email, you need to open port 40405 since develop to access webscocket from admin.

Can you explain more "but devices are not updated some time in admin interface.". I don't understand.

The plugin weather does not work with your device (empty response from yahoo) but works with paris.

About the plugins startup, if you configure the plugins to start with domogik, they start... tested with weather and ping.

About domoweb that does not see the devices, you are true. I also reproduce it on your server. Can you open a related issue on domoweb side ?

So, to sum up, I don't see any issue on domogik. I need more explanations

**Comment by vdomos** [comment]
I haven't open 40405 port for external acces. That's why it did not work for you

Port 40405 is non blocked in local, there is no firewall.

Port is listen en all interfaces:

    tcp        0      0 10.0.2.15:40405         0.0.0.0:*               LISTEN      5980/python     
    tcp        0      0 192.168.0.31:40405      0.0.0.0:*               LISTEN      5980/python     
    tcp        0      0 127.0.0.1:40405         0.0.0.0:*               LISTEN      5980/python

I can telnet on this port from the nagivator's host.

For "the devices are not updated some time in admin interface.":
I test with ping plugin and "google dns" device, you add. In the plugin's log, it's ok 

    2017-05-11 23:20:43,296 domogik-ping DEBUG 8.8.8.8 is alive
    2017-05-11 23:20:43,297 domogik-ping DEBUG Send xpl message...
    2017-05-11 23:20:43,298 domogik-ping DEBUG xpl-stat
    {
    hop=1
    source=
    target=
    }
    sensor.basic
    {
    device=8.8.8.8
    current=high
    type=input
    }

in my xpl log, it's ok:

    2017-05-11 23:20:43  xpl-stat { hop=1 source=domogik-ping.ares target=* } sensor.basic { device='8.8.8.8' current='high' type='input' }

but in admin interface, the "Last value" is not display and in mysql table, there is no value store.

    MariaDB [domogik]> select id, name, last_value from core_sensor where device_id=19 ;
    +-----+------+------------+
    | id  | name | last_value |
    +-----+------+------------+
    | 246 | Ping | NULL       |
    +-----+------+------------+
    1 row in set (0.00 sec)

**Comment by vdomos** [comment]
After restarting Domogik, the sensor is up to date.

This seems to happen after adding a new device, update in the database does not work.
Same thing when restarting the plugin.

**Comment by fritz-smh** [comment]
Can you check if this is better with the last commits ?

**Comment by vdomos** [comment]
Sorry, there is still some issue like:
- Plugins takes >~10s for starting in admin interface.

- New device create are not update in admin interface and database:
Example: 
With a stop plugin (onewired),
Add new device.
Start plugin, old devices are update but not the new one.

New device sensor id=280
Log onewired:

    2017-05-30 22:35:49,369 domogik-onewired INFO ==> Reading sensor '/28.2CAED7010000/temperature'
    2017-05-30 22:35:50,054 domogik-onewired DEBUG ==> Update Sensor '{280: '15.6'}' for device id 37 (Extèrieure Ow)

Log core_xplgw (Traceback,  the same like above):

    core_xplgw.log:2017-05-30 22:35:50,055 domogik-xplgw DEBUG New message (from MQ) > start to store each of the sensors in the store queue...
    core_xplgw.log:2017-05-30 22:35:50,055 domogik-xplgw DEBUG New message (from MQ) > message for sensor_id='280' added to the store queue, current length = 1
    core_xplgw.log:2017-05-30 22:35:50,055 domogik-xplgw DEBUG New message (from MQ) > storing in the store queue finished
    core_xplgw.log:2017-05-30 22:35:50,099 domogik-xplgw DEBUG Getting item from the store queue, current length = 0, item = '{'sensor_id': u'280', 'value': u'15.6', 'time': 1496176550}'
    core_xplgw.log:2017-05-30 22:35:50,100 domogik-xplgw ERROR Traceback (most recent call last):

In database, no value too:

    MariaDB [domogik]> select id,device_id,name,last_received,last_value from core_sensor where id=280;
    +-----+-----------+-------------+---------------+------------+
    | id  | device_id | name        | last_received | last_value |
    +-----+-----------+-------------+---------------+------------+
    | 280 |        37 | Temperature |          NULL | NULL       |
    +-----+-----------+-------------+---------------+------------+

The same with oher plugin.


- Other stange issue:

Plugin onewired is configured in database:

     MariaDB [domogik]> select * from core_plugin_config where id="onewired";
     +----------+--------+----------+---------------+------------+
     | id       | type   | hostname | key           | value      |
     +----------+--------+----------+---------------+------------+
     | onewired | plugin | hades    | 1-wire_cache  | Y          |
     | onewired | plugin | hades    | 1-wire_device | vesta:4304 |
     | onewired | plugin | hades    | auto_startup  | N          |
     | onewired | plugin | hades    | configured    | True       |
     +----------+--------+----------+---------------+------------+

but not in admin ...

![issue_1](https://cloud.githubusercontent.com/assets/1257481/26607103/2953534e-4595-11e7-8593-f9402ef413ed.png)



I test this on 2 differents Vms with 1G and 2Go of Ram, the cpus are not overloaded.

**Comment by jschaeke** [comment]
similar issues here since I updated yesterday to develop. I can't restart domogik as I have to kill the gunicorn first (otherwise admin wouldn't be alive upon start). Even when the admin is alive not all plugins are started. A couple of start/stops of domogik with killing gunicorns in between helps all plugins to be available again in the admin. But the mqtt plugin looses its configuration and I have to complete it manually and stop/start it a couple of times the plugin itself to be alive.

**Comment by jschaeke** [comment]
Seems the last commits did improve the startup situation here (everything starts). Killing gunicorn is still required when restarting domogik.

**Comment by vdomos** [comment]
it's ok for me with last Domogik updates.
A fresh install with installation script works fine (with domogik-mq develop)
